src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java

```diff
@@ -766,6 +839,12 @@ public boolean removePlayer(final UUID id)
         if (player != null && player.getRank() != Rank.OWNER && players.remove(id) != null)
         {
             markDirty();
+
+            if (getOwnerName().equals("[abandoned]") && getPlayersByRank(Rank.OFFICER).isEmpty())
```

Comment:
maybe this if here could be in general a method we call in all the method

Reply:
I mean its very short and only called three times, but ok i will move it to its own method.